exodus/trackers/views.py

```diff
@@ -21,13 +21,25 @@ def index(request):
 
 
 def detail(request, tracker_id):
-    reports_number = Report.objects.count()
+    # Count of applications(similar to getting count of one report per application)
+    reports_number = Application.objects.distinct('handle').count()
     try:
         tracker = Tracker.objects.get(pk=tracker_id)
         # Add spaces aroung pipes for better rendering of signatures
         tracker.network_signature = tracker.network_signature.replace("|", " | ")
         tracker.code_signature = tracker.code_signature.replace("|", " | ")
-        reports_list = Report.objects.order_by('-creation_date').filter(found_trackers=tracker_id)
+
+        # Returns reports in reverse chronological order
+        all_reports = Report.objects.order_by('-creation_date').filter(found_trackers=tracker_id)
+        # List of only latest report for an application
+        reports_list = []
+        application_map = {}
+        for report in all_reports:
+            app_handle = report.application.handle
+            if app_handle not in application_map:
+                application_map[app_handle] = True
+                reports_list.append(report)
```

Comment:
If I'm correct, this does not exactly what we want.
For instance, if a given tracker is present in v1 of an application but is not anymore in v2 of the same application, we don't want to list it here.

Reply:
@simpnu the order of reports is the latest first. So if the tracker is not part of an application it won't even be present in the list as we're filtering based on the tracker_id. 
Correct me if I'm missing something here.

Reply:
Oh I got it. I'll fix this.

Reply:
@simpnu I've updated the code and added a new test case to check this functionality as well. Please check it now